src/coreComponents/common/getAvailableMemory.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef GET_AVAILABLE_MEMORY_HPP
+#define GET_AVAILABLE_MEMORY_HPP
+
+#include <sys/stat.h>
+#if defined __MACH__
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+#include <mach/host_info.h>
+#include <mach/mach_host.h>
+#include <mach/task_info.h>
+#include <mach/task.h>
+#endif
+
+namespace geos
+{
+/**
+ * @brief Retieves current available memory on host
+ * @return the available memory in bytes.
+ */
+static inline size_t getAvailableMemory()
```

Comment:
Maybe indicate in the function name that this is for host/system memory?

Reply:
Good idea.